### ISSUE TITLE
Loading logic needs to be completely redone

### ISSUE THREAD

**Issue by TrikkStar** [TrikkStar/WebApplications_Final#22]
Files are not being loaded in the proper way, cannot trace the call which is causing the whole thing to crash while loading.

**Comment by TrikkStar** [comment]
Problem is being caused by the passing of the `Algorithms` object to the template.

**Comment by TrikkStar** [comment]
@skiadas, I found what was calling my algorithms object and not passing it any parameters. My question is why is it being called when I supply it to the template?

**Comment by skiadas** [comment]
Well let's think of what you are asking it to do. It runs with `Algorithms` as the context.

Then you ask it for `{{#each algArray}}`

So it will try to go through your `algArray`, which was defined thus:

```
	Algorithms.algArray = [Algorithms.selection, Algorithms.insertion, Algorithms.mergesort, Algorithms.quicksort];
```

So your algArray is an array of functions. So it goes through each of these, and you ask it to do:

```
<option value="{{this}}">{{this}}</option>
```
So you are asking it to somehow take "this", i.e. the function `Algorithms.selection`, and produce a string from it. I guess the way it tries to do that is by calling the function.

In other words, the array you use in your template should give the string names for your algorithms, not the algorithms themselves.
